modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/slot/SlotDeletingTask.java

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright (c) 2017, WSO2 Inc. (http://wso2.com) All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.wso2.andes.kernel.slot;
+
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.andes.kernel.MessagingEngine;
+import org.wso2.andes.kernel.subscription.StorageQueue;
+
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Task executing the slot delete. It will poll on slots scheduled to delete and
+ * coordinate with slot coordinator in cluster to delete slots.
+ */
+public class SlotDeletingTask implements Runnable {
+
+    private static Log log = LogFactory.getLog(SlotDeletingTask.class);
+
+    /**
+     * Condition running the task.
+     */
+    private boolean isLive = true;
+
+    /**
+     * Queue to keep slots to be removed. Reason to use a queue is
+     * if a slot could not be removed, it is definite later slots cannot be removed
+     * as well (because of safe-zone)
+     */
+    private LinkedBlockingQueue<Slot> slotsToDelete;
```

Comment:
when we have both slow and fast consumers for 2 queues, shouldn't we try to delete the next slot instead of waiting until the earliest slot clears up ?